### PR TITLE
adding proposal api support + resource api

### PR DESCRIPTION
**Pull request by kookiekrak**
Adds SR proposal api to tronweb.

Also adds a missing api for pulling user resources.

Built dist files.

### CODE REVIEW

File: src/lib/transactionBuilder.js [hunk 1]
```diff
@@ -652,4 +652,104 @@ export default class TransactionBuilder {
...
+        if(!callback)
+            return this.injectPromise(this.voteProposal, proposalVoterAddress, proposalID, isApprovalDecision);
+
+        if(!this.tronWeb.isAddress(proposalVoterAddress))
+            return callback('Invalid proposalVoterAddress address provided');
+
+        if(!utils.isInteger(proposalID) || proposalID < 0)
+            return callback('Invalid proposalID provided');
+
+        if(!utils.isString(isApprovalDecision))
```
[the remark this answers is not in the document]
Comment (Kondax) [reply src/lib/transactionBuilder.js]: Ah in which case just do `utils.isBoolean(isApprovalDecision)`, and when using it in the http parameter `isApprovalDecision.toString()`

File: src/lib/transactionBuilder.js [hunk 2]
```diff
@@ -652,4 +652,162 @@ export default class TransactionBuilder {
...
+        if(!utils.isString(tokenName) || !tokenName.length)
+            return callback('Invalid tokenName provided');
+
+        if(!utils.isInteger(tokenAmount) || tokenAmount < 0)
+            return callback('Invalid tokenAmount provided');
+
+        this.tronWeb.fullNode.request('wallet/exchangeinject', {
+            owner_address: this.tronWeb.address.toHex(ownerAddress),
+            exchange_id: parseInt(exchangeID),
+            token_id: tokenName,
```
Comment (Kondax) [inline src/lib/transactionBuilder.js]: Does the API expect the name in HEX or Utf8? If it's HEX, use `this.tronWeb.fromUtf8(exchangeID)`

File: src/lib/transactionBuilder.js [hunk 3]
```diff
@@ -652,4 +652,162 @@ export default class TransactionBuilder {
...
+        if(!this.tronWeb.isAddress(ownerAddress))
+            return callback('Invalid ownerAddress provided');
+
+        if(!utils.isInteger(exchangeID) || exchangeID < 0)
+            return callback('Invalid exchangeID provided');
+
+        if(!utils.isString(tokenName) || !tokenName.length)
+            return callback('Invalid tokenName provided');
+
+        if(!utils.isInteger(tokenAmount) || tokenAmount < 0)
```
Comment (Kondax) [inline src/lib/transactionBuilder.js]: I have a feeling there will be at least 1 token required

**Comment by kookiekrak** [comment]
i fucked it up. closing this

**Pull request closed without merging**